File: #003 Sieve of Eratosthenes/sieve.py
import sys
import math

def sieve(n):
    a = []
    for i in range(n+1):
        a.append(True)
    
    for i in range(2,int(math.sqrt(n+1))):
        if a[i]:
            j = i*i
            while j <= n:
                a[j] = False
                j +=i
    
    return a

def main():
    n = int(sys.argv[1])
    a = sieve(n)
    primes = []
    for i in range(2, n+1):
        if a[i]:
            primes.append(i)
    print(primes)

if __name__=="__main__":
    main()

File: #003 Sieve of Eratosthenes/sieve.java
public class sieve{
    public static boolean[] sieve(int n){
        boolean[] a = new boolean[n+1];
        for(int i=0;i<a.length;i++){
            a[i] = true;
        }

        for(int i=2;i<=Math.sqrt(n);i++){
            if(a[i] == true){
                System.out.println("i: " + String.valueOf(i));
                for(int j=i*i; j<=n; j=j+i){
                    System.out.println("j: " + String.valueOf(j));
                    a[j] = false;
                }
            }
        }

        return a;
    }

    public static void main(String[] args){
        int n = Integer.parseInt(args[0]);
        boolean[] a = sieve(n);
        String prefix = "";
        System.out.print("[");
        for(int i=2;i<a.length;i++){
            if(a[i] == true){
                System.out.print(prefix);
                System.out.print(String.valueOf(i));
                prefix = ", ";
            }
        }
        System.out.print("]");
    }
}